File: src/test/console/java/src/test/java/kobiton/com/testng/iOSWebTest.java
package kobiton.com.testng;

import com.kobiton.service.DeviceService;
import com.kobiton.service.model.Device;
import java.util.concurrent.TimeUnit;
import org.testng.Assert;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;
import java.io.IOException;
import kobiton.com.utils.AutomationUtils;
import org.openqa.selenium.remote.RemoteWebDriver;

public class iOSWebTest {

    public static RemoteWebDriver driver = null;
    String successMsg = "You logged into a secure area!";

    @BeforeTest
    public void Setup() {
        try {
            Device device = new DeviceService().getOnlineDevice("iOS");
            System.out.println(String.format("Execute with: %s  udid: %s", device.getDeviceName(), device.getUdid()));
            driver = new RemoteWebDriver(AutomationUtils.kobitonServerUrl(), AutomationUtils.desiredCapabilitiesiOSWeb(device));
            driver.manage().timeouts().implicitlyWait(60, TimeUnit.SECONDS);
            String kobitonSessionId = driver.getCapabilities().getCapability("kobitonSessionId").toString();
            System.out.println("https://portal.kobiton.com/sessions/" + kobitonSessionId);
        } catch (IOException ex) {
            Assert.fail("set up failed", ex);
        }

    }

    @AfterTest
    public void Teardown() {
        try {
            if (driver != null) {
                driver.quit();
            }
        } catch (Exception e) {
            //Do-nothing
        }
    }

    @Test(priority = 1, description = "should run test successfully with correct username and password")
    public void testLoginSuccessfully() {
        login("tomsmith", "SuperSecretPassword!");
        Assert.assertTrue(getMessage().contains(successMsg));
    }

    public void login(String userName, String password) {
        driver.get("http://the-internet.herokuapp.com/login");
        driver.findElementById("username").sendKeys(userName);
        driver.findElementById("password").sendKeys(password);
        driver.findElementByXPath("//form[@name='login']").submit();
    }

    public String getMessage() {
        return driver.findElementById("flash").getText();
    }
}

File: src/test/console/dotnet/MobileAutomation/Tests/Scripts/IOSWebTest.cs
﻿using System;
using MobileAutomation.Config;
using MobileAutomation.Service;
using MobileAutomation.Service.Model;
using MobileAutomation.Tests.Utils;
using OpenQA.Selenium.Remote;
using Xunit;

namespace MobileAutomation.Tests.Scripts
{
    public class IOSWebTest : BaseTest
    {
        string successMsg = "You logged into a secure area!";

        public IOSWebTest()
        {
            var device = new DeviceService().GetOnlineDeviceSync(DeviceType.IOS);
            Console.WriteLine(string.Format("Device {0} osVersion: {1} udid: {2}",
                                            device.DeviceName,
                                            device.PlatformVersion,
                                            device.Udid));

            driver = new RemoteWebDriver(AutomationUtils.Instance.KobitonServerUrl,
                                        AutomationUtils.Instance.DesiredCapabilitiesIOSWeb(device),
                                        TimeSpan.FromSeconds(Configs.SESSION_TIMEOUT));
            driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(60);
        }

        [Fact]
        public void TestIFixitiOSWeb()
        {
            TestLoginSuccessfully();
        }

        public void TestLoginSuccessfully()
        {
            Console.WriteLine("should run test successfully with correct username and password");
            Login("tomsmith", "SuperSecretPassword!");
            Assert.True(GetMessage().Contains(successMsg));
        }

        private void Login(String userName, String password)
        {
            driver.Navigate().GoToUrl("http://the-internet.herokuapp.com/login");
            driver.FindElementById("username").SendKeys(userName);
            driver.FindElementById("password").SendKeys(password);
            driver.FindElementByXPath("//form[@name='login']").Submit();
        }

        private String GetMessage()
        {
            return driver.FindElementById("flash").Text;
        }
    }
}
